### PR TITLE
tools: refactor faq and troubleshooting for DM

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What is changed, added or deleted? 

refactor faq and troubleshooting for DM

FAQ should be some question like "can DM do something?"
troubleshooting should be "when DM meet some trouble, how to resolve it"

### Which version does your change affect? 

only update dev now, will update v2.1 and v3.0 after get two LGTM

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@csuzhangxc PTAL

**Comment by WangXiangUSTC** [comment]
> I suggest keeping FAQ and troubleshooting as two separate sections in TOC.

I update in cf9e3d8, PTAL again

File: dev/TOC.md [hunk 1]
```diff
@@ -104,9 +103,11 @@
       + 从与 MySQL 兼容的数据库迁移数据
         - [从 Amazon Aurora MySQL 迁移数据](/dev/how-to/migrate/from-aurora.md)
       - [DM Portal](/dev/reference/tools/data-migration/dm-portal.md)
-      - [DM 错误含义和诊断](/dev/reference/tools/data-migration/error-system.md)
-      - [DM 故障诊断](/dev/how-to/troubleshoot/data-migration.md)
-      - [FAQ](/dev/faq/data-migration.md)
+      + DM 故障诊断
+        - [DM 故障诊断](/dev/reference/tools/data-migration/faq-and-troubleshooting/troubleshooting.md)
```
Comment (lilin90) [inline dev/TOC.md]: Avoid using a deep directory.

```suggestion
        - [DM 故障诊断](/dev/reference/tools/data-migration/troubleshooting.md)
```

File: dev/TOC.md [hunk 2]
```diff
@@ -104,9 +103,11 @@
...
+        - [DM 错误含义和诊断](/dev/reference/tools/data-migration/faq-and-troubleshooting/error-system.md)
```
Comment (lilin90) [inline dev/TOC.md]:
```suggestion
        - [DM 错误含义和诊断](/dev/reference/tools/data-migration/error-system.md)
```

File: dev/TOC.md [hunk 3]
```diff
@@ -104,9 +103,11 @@
...
+        - [DM 常见错误及修复方法](/dev/reference/tools/data-migration/faq-and-troubleshooting/error-handler.md)
```
Comment (lilin90) [inline dev/TOC.md]:
```suggestion
        - [DM 常见错误及修复方法](/dev/reference/tools/data-migration/error-handler.md)
```

File: dev/reference/tools/data-migration/faq-and-troubleshooting/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+title: Data Migration 故障诊断
+category: reference
```
Comment (lilin90) [inline dev/reference/tools/data-migration/faq-and-troubleshooting/troubleshooting.md]:
```suggestion
category: reference
aliases: ['/docs-cn/dev/how-to/troubleshoot/data-migration/']
```

File: dev/reference/tools/data-migration/faq.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+---
+title: Data Migration 常见问题
+category: reference
```
Comment (lilin90) [inline dev/reference/tools/data-migration/faq.md]:
```suggestion
category: reference
aliases: ['/docs-cn/dev/faq/data-migration/']
```

File: dev/reference/tools/data-migration/faq-and-troubleshooting/error-handler.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+---
+title: Data Migration 常见错误及修复
+category: reference
+---
+
+# Data Migration 常见错误及修复
+
```
Comment (lilin90) [inline dev/reference/tools/data-migration/faq-and-troubleshooting/error-handler.md]: Please add a brief intro to this doc using one or two sentences.

File: dev/reference/tools/data-migration/faq-and-troubleshooting/error-system.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Data Migration 错误说明和诊断
+title: Data Migration 错误说明
 category: reference
 ---
 
-# Data Migration 错误说明和诊断
+# Data Migration 错误含义
```
Comment (lilin90) [inline dev/reference/tools/data-migration/faq-and-troubleshooting/error-system.md]: Please keep the 1st-level heading consistent with the `title` in metadata.

**Comment by WangXiangUSTC** [comment]
@lilin90 thanks, all comment resolved, PTAL again

**Comment by lilin90** [comment]
@csuzhangxc @WangXiangUSTC I rename some files to make the name shorter, and make changes to v3.0 and v2.1. PTAL~

**Comment by WangXiangUSTC** [comment]
@lilin90 thanks, LGTM

**Pull request merged**